### PR TITLE
Allow SoftSignals to have and propagate metadata

### LINKED ISSUE #356: Allow SoftSignals metadata (units, limits etc.)
SoftSignals are being used to propagate metadata about devices, such as the dodal Undulator length. length requires a unit (metres/m). Other soft signals may require limits. These would (after #137, #146, #148) propagate into documents.

### PR DESCRIPTION
**Pull request by DiamondJoseph**
Closes #356

### CODE REVIEW

**Review by coretl — APPROVED** [review]

**Comment by coretl** [comment]
when test is added...

File: src/ophyd_async/core/soft_signal_backend.py [hunk 1]
```diff
@@ -109,14 +142,20 @@ class SoftSignalBackend(SignalBackend[T]):
...
     _severity: int
+    _units: Optional[str]
+    _precision: Optional[int]
 
     def __init__(
         self,
         datatype: Optional[Type[T]],
         initial_value: Optional[T] = None,
+        units: Optional[str] = None,
+        precision: Optional[int] = None,
```
Comment (DiamondJoseph) [inline src/ophyd_async/core/soft_signal_backend.py:153]: Or SoftSignalBackend can take kwargs of other fields to pass as configuration.
e.g. units, precision, choices, limits

**Comment by DiamondJoseph** [comment]
Replaced by #367

**Comment by DiamondJoseph** [comment]
Wrong branch

**Pull request merged**